TheJumpCloud/support
# tests/ error output in blanket catch statement

## Issues
* [SA-3356](https://jumpcloud.atlassian.net/browse/SA-3356) - Add API error messages to Update-JCUsersFromCSV

## What does this solve?

If using the PowerShell Module to update users from CSV, caught error messages were not being returned correctly. Instead, a blanket statement "User does not exist" error was being thrown. This was confusing and incorrect especially since the error would be thrown even in the intended user existed. Now the function will return the caught errors from our API.

## Is there anything particularly tricky?

NO

## How should this be tested?

Automated tests are included in the PR but this can be tested locally:

In the current module version run this code:
```pwsh
$CSVDATA = @{
                Username         = "someUsernameInYourOrg"
                Attribute1_name  = "bad value"
                Attribute1_value = 'string'
}
$CSVFILE = $CSVDATA | Export-Csv "./custom_attribute.csv" -Force
$UpdateStatus = Update-JCUsersFromCSV -CSVFilePath "./custom_attribute.csv" -force
```
The module should return an incorrect error stating that the user does not exist. Import this module version and re-run the code. the correct error for the custom attribute name should be caught and displayed.

## Screenshots

<img width="535" alt="Screen Shot 2023-05-26 at 11 51 40 AM" src="https://github.com/TheJumpCloud/support/assets/54448601/adf1d990-1a3e-4186-8c94-8085d2f9eeaf">


[SA-3356]: https://jumpcloud.atlassian.net/browse/SA-3356?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
